src/utils/hooks/useGetResponsesByCode.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState, useEffect } from 'react'
+import {
+  Application,
+  ApplicationResponse,
+  ApplicationSection,
+  useGetApplicationQuery,
+  GetApplicationQuery,
+} from '../generated/graphql'
+import { ResponsesByCode } from '../types'
+
+interface useLoadApplicationProps {
+  serialNumber: string
+}
+
+const useGetResponsesByCode = (props: useLoadApplicationProps) => {
+  const { serialNumber } = props
+  const [responsesByCode, setResponsesByCode] = useState({})
+  const [error, setError] = useState('')
+  const { data, loading: apolloLoading, error: apolloError } = useGetApplicationQuery({
+    variables: { serial: serialNumber },
+  })
+
+  useEffect(() => {
+    const error = checkForApplicationErrors(data)
+
+    if (error) {
+      setError(error)
+      return
+    }
+
+    const applicationResponses = data?.applications?.nodes[0]?.applicationResponses
+      .nodes as ApplicationResponse[]
+
+    const currentResponses = {} as ResponsesByCode
+
+    applicationResponses.forEach((response) => {
```

Comment:
This line fails if I try accessing the address of one existing application page with the **server not running**: e.g: ` http://localhost:3000/applications/1236/SectionA/page1`. 
I started doing that to make sure we are covering cases of bad responses from the server. I think this is another case to be checked on the `checkForApplicationErrors` or perhaps just returning the `apolloError` might fix that. See what I've done on `ApplicationCreate`.

Reply:
Created a separated issue for that: #60